python-package/pyproject.toml

```
@@ -30,7 +30,7 @@ maintainers = [
 name = "lightgbm"
 readme = "README.rst"
 requires-python = ">=3.7"
-version = "4.4.0.99"
+version = "4.4.0"
```

I guess it was a typo.
```suggestion
version = "4.5.0"
```

---

ah!! yes, thank you. Maybe in the future we should have a script that automates these changes, like my colleagues do in the RAPIDS repos: https://github.com/rapidsai/cuml/blob/branch-24.08/ci/release/update-version.sh

---

Great idea!

---

Very sorry for resolving this without answering you @borchero , that was unintentional.

I'm -1 on using tools like `setuptools-scm` or `versioneer` here. In my experience, they introduce much more complexity than they reduce, at least for projects like LightGBM with a linear sequence of releases (no long-term-support branches, backporting, etc.).

For example, they can be the source of hard-to-debug differences between local and CI builds, unless you're diligent about running `git fetch upstream --tags` or similar regularly.

You might find some of the other discussion in https://discuss.python.org/t/please-make-package-version-go-away/58501 interesting (although I don't agree with the original post... I believe we should continue to support `lightgbm.__version__`).